### DISCUSSION TITLE
Webhook based auto scaling not working with EKS

### DISCUSSION THREAD

**Discussion by SejalParmar8796** [actions/actions-runner-controller#1677] [Questions]
Hi,
I have setup the ARC in EKS cluster and type of EC2 node. I have run the helm script for webhook server creation which is mentioned in document. After that if I try to create a webhook in github and test, it says either 400 or 404. Also I tried running jobs to check whether it scales up or not based on the event. But it is not scaling up. Can someone please help me to implement this ?

**Comment by ViktorLindgren95** [comment]
If you runners can still register to github or another private enviroment but the webhook cant reach you, then its inbound traffic. you might have to create some form of opening in your network etc to allow inbound traffics for the webhook.

**Reply by SejalParmar8796** [reply]
I have checked my security group and also allowed all the traffic from internet in the security group. but still facing the same issue. :(

**Comment by jishaashok** [comment]
Hi there, I have been trying to get the webhook working for almost 2 weeks now. Can you please share step-by-step instructions on how to set up the load balancer to work with the ingress. 

I am trying to set up webhook server with ingress. What are the configurations I need to use? On AWS console, how do I set up the load balancer to route to the ingress? 

kubectl describe svc actions-runner-controller-webhook --namespace actions-runner-system
Name:              actions-runner-controller-webhook
Namespace:         actions-runner-system
Labels:            app.kubernetes.io/instance=actions-runner-controller
                   app.kubernetes.io/managed-by=Helm
                   app.kubernetes.io/name=actions-runner-controller
                   app.kubernetes.io/version=0.25.2
                   helm.sh/chart=actions-runner-controller-0.20.2
Annotations:       meta.helm.sh/release-name: actions-runner-controller
                   meta.helm.sh/release-namespace: actions-runner-system
Selector:          app.kubernetes.io/instance=actions-runner-controller,app.kubernetes.io/name=actions-runner-controller
Type:              ClusterIP
IP Family Policy:  SingleStack
IP Families:       IPv4
IP:                10.100.213.4
IPs:               10.100.213.4
Port:              https  443/TCP
TargetPort:        9443/TCP
Endpoints:         192.168.29.xx:9443
Session Affinity:  None
Events:            <none>


Thank you,

**Reply by SejalParmar8796** [reply]
Hi,
I have created the internal load balancer which is the option available in AWS.
Once you install ARC, after that you need to run the following helm command.
helm upgrade --install --namespace actions-runner-system --create-namespace --wait actions-runner-controller actions-runner-controller/actions-runner-controller --set "githubWebhookServer.enabled=true,service.type=NodePort,githubWebhookServer.service.type=LoadBalancer"

Once this is completed, you can check the URL of the load balancer using the command:
kubectl get svc -n actions-runner-system | grep "LoadBalancer"

You have to get that URL and set it in github webhook with http://<URL> and content type should be application/json
